File: frontend/src/objects/merchant/store/StoreOperations.ts
// Business note: frontend mirror of a backend service object; keep names and fields aligned unless this file is explicitly frontend-only.
import type {
  AddressText,
  ImageUrl,
  Minutes,
  StoreStatus,
} from '@/objects/core/SharedObjects'
import type { MenuItem } from '@/objects/merchant/menu/MenuItem'
import type { BusinessHours } from '@/objects/merchant/store/BusinessHours'
import type { StoreLocation } from '@/objects/merchant/store/StoreLocation'

export type StoreOperations = {
  status: StoreStatus
  storeAddress: AddressText
  location?: StoreLocation
  businessHours: BusinessHours
  avgPrepMinutes: Minutes
  imageUrl?: ImageUrl
  menu: MenuItem[]
}

File: backend/src/main/scala/services/merchant/objects/StoreOperations.scala
package services.merchant.objects

// Business note: service-owned domain object; mirror it in the frontend only when it is part of the protocol surface.
import system.objects.given
import services.customer.objects.*

import system.objects.*
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.*

final case class StoreOperations(
    status: DisplayText,
    storeAddress: AddressText,
    location: Option[StoreLocation],
    businessHours: BusinessHours,
    avgPrepMinutes: Minutes,
    imageUrl: Option[ImageUrl],
    menu: List[MenuItem],
)
object StoreOperations:
  given Encoder[StoreOperations] = deriveEncoder
  given Decoder[StoreOperations] = deriveDecoder
